03-string-object/hw-03-string-object.py

```diff
@@ -0,0 +1,46 @@
+import re
+import string
+
+text_to_fix = """homEwork:
+  tHis iz your homeWork, copy these Text to variable.
+
+
+  You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each existING SENtence and add it to the END OF this Paragraph.
+
+
+  it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+
+
+  last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+"""
+# fixed "iz" words
+fixed_text = re.sub(r'(\biz\b)', 'is', text_to_fix, flags=re.I)
```

Comment:
Using regex is hard way to do this task
Also in output there is """**Fix“is” with correct “is”**"""
Suppose that first "iz" should remain as "iz"

Better code is to split text to sentences by '.' and than sentences to words
My example:

_**for sentence in list_input_sentences:
    modified_sentence = sentence.strip().lower().capitalize()
    list_output_words = []
    for word in modified_sentence.split(' '):
        if word.lower() == "iz":
            word = word.replace("z", "s")
        list_output_words.append(word)
    modified_sentence = " ".join(list_output_words)
    list_output_sentences.append(modified_sentence)
    list_extra_sentence_words.append(list_output_words[-1])
extra_sentence = " ".join(list_extra_sentence_words).capitalize()
list_output_sentences.append(extra_sentence)**_